sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DistributionAndOrderingUtils.scala

```diff
@@ -83,13 +83,17 @@ object DistributionAndOrderingUtils {
         queryWithDistribution
       }
 
-      // Apply typeCoercionRules since the converted expression from TransformExpression
-      // implemented ImplicitCastInputTypes
-      typeCoercionRules.foldLeft(queryWithDistributionAndOrdering)((plan, rule) => rule(plan))
+      ResolveTimezoneAndCastExecutor.execute(queryWithDistributionAndOrdering)
```

Comment:
shall we generate expressions with timezone specified?

Reply:
I am not sure I fully understand the question.

The `TimeZoneAwareExpression` may come from the implicitly Cast(see the case in the PR description), not only the v2 function construction(which I think I can control how to construct the expression and set timezone explicitly).

Reply:
oh I see